src/steamship/agents/llm.py

```diff
@@ -0,0 +1,36 @@
+from abc import ABC, abstractmethod
```

Comment:
A few thoughts on the `LLM` abstraction above merely `use_plugin`:

* I think it'll serve us really well in the Agent & Tools platform because it will help us curate a more "it always works" collection of options over-top the underlying plugin infra.
* I even think we might find ourselves (downstream..) wanting to have the defaults stored in some `settings` file in the Workspace that the user can override via API / Web Interface. E.g. you should be able to call the `switch_llm()` method that changes from gpt-3.5 to 4 to cohere, etc etc
* It also seems like a good way to enforce some kind of tracing / agent-specific logging.

Basically a long-winded way to say: two thumbs up to this direction from my end!